File: VB/dxSampleGrid/MyClasses/MyViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Linq
Imports System.Runtime.CompilerServices
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Input

Namespace dxSampleGrid
	Public Class MyViewModel

		Public Sub New()
			CreateList()
			ViewModelName = "MyViewModel1"
			SomeValue = 5
		End Sub
		Private _viewModelName As String
		Private _selectedPerson As Person
		Private _myVoidCommand As ICommand


		Private privateListPerson As ObservableCollection(Of Person)
		Public Property ListPerson() As ObservableCollection(Of Person)
			Get
				Return privateListPerson
			End Get
			Set(ByVal value As ObservableCollection(Of Person))
				privateListPerson = value
			End Set
		End Property
		Public Property ViewModelName() As String
			Get
				Return _viewModelName
			End Get
			Set(ByVal value As String)
				_viewModelName = value

			End Set
		End Property
		Private privateSomeValue As Integer
		Public Property SomeValue() As Integer
			Get
				Return privateSomeValue
			End Get
			Set(ByVal value As Integer)
				privateSomeValue = value
			End Set
		End Property
		Public Property SelectedPerson() As Person
			Get
				Return _selectedPerson
			End Get
			Set(ByVal value As Person)
				_selectedPerson = value

			End Set
		End Property
		Private privateListSomeClass As ObservableCollection(Of SomeClass)
		Public Property ListSomeClass() As ObservableCollection(Of SomeClass)
			Get
				Return privateListSomeClass
			End Get
			Set(ByVal value As ObservableCollection(Of SomeClass))
				privateListSomeClass = value
			End Set
		End Property

		Public ReadOnly Property MyVoidCommand() As ICommand
			Get
				'if (_myVoidCommand == null)
				'    _myVoidCommand = new DelegateCommand(SomeMethod);
				Return _myVoidCommand
			End Get
		End Property


		Private Sub CreateList()
			ListPerson = New ObservableCollection(Of Person)()
			For i As Integer = 0 To 9
				Dim p As New Person(i)
				ListPerson.Add(p)
			Next i
			SelectedPerson = ListPerson(3)
			ListSomeClass = New ObservableCollection(Of SomeClass)()
			For i As Integer = 0 To 199 Step 10
				ListSomeClass.Add(New SomeClass(i))
			Next i
		End Sub
		Private Sub SomeMethod()
			Debug.Print("Some action")

		End Sub

		


	End Class
End Namespace

File: CS/dxSampleGrid/MyClasses/MyViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Diagnostics;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Input;

namespace dxSampleGrid {
    public class MyViewModel : INotifyPropertyChanged {
        public MyViewModel() {
            CreateList();
            ViewModelName = "MyViewModel1";
            SomeValue = 5;
        }
        string _viewModelName;
        Person _selectedPerson;
        ICommand _myVoidCommand;


        public ObservableCollection<Person> ListPerson { get; set; }
        public string ViewModelName {
            get { return _viewModelName; }
            set {
                _viewModelName = value;
                RaisePropertyChanged();
            }
        }
        public int SomeValue { get; set; }
        public Person SelectedPerson {
            get { return _selectedPerson; }
            set {
                _selectedPerson = value;
                RaisePropertyChanged();
            }
        }
        public ObservableCollection<SomeClass> ListSomeClass { get; set; }

        public ICommand MyVoidCommand {
            get {
                //if (_myVoidCommand == null)
                //    _myVoidCommand = new DelegateCommand(SomeMethod);
                return _myVoidCommand;
            }
        }


        void CreateList() {
            ListPerson = new ObservableCollection<Person>();
            for (int i = 0; i < 10; i++) {
                Person p = new Person(i);
                ListPerson.Add(p);
            }
            SelectedPerson = ListPerson[3];
            ListSomeClass = new ObservableCollection<SomeClass>();
            for (int i = 0; i < 200; i += 10)
                ListSomeClass.Add(new SomeClass(i));
        }
        void SomeMethod() {
            Debug.Print("Some action");

        }

        public event PropertyChangedEventHandler PropertyChanged;
        private void RaisePropertyChanged([CallerMemberName]String propertyName = "") {
            if (PropertyChanged != null) {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }


    }
}
